### ISSUE TITLE
[0.19.x] queryChaincode wont try next peer if peer is stopped or deleted (IBM Cloud)

### ISSUE THREAD

**Issue by crazyman1979** [hyperledger/composer#4412]
I have composer rest server hitting my channel which has multiple peers, when i disable one of the peers it doesn't try the next one.

Im thinking of putting in a PR with this fix below.

## Context
`hlfQueryHandler.queryChaincode` has retry logic based on `querySinglePeer` throwing an exception when the peer is unreachable, this doesn't happen when i disable the peer in IBM cloud.

## Expected Behavior
`queryChainCode` should catch exception thrown by querySinglePeer when peer is down, then try the next peer in the list.

## Actual Behavior
No exception is thrown by `querySinglePeer` because it only throws if the exception has a code of 14 - UNAVAILABLE

Logs shows 

**2018-09-25T13:23:27.982Z DEBUG   :HLFConnectionManager     :fabric-client()           [Peer.js]: Received proposal response from: grpcs://ne4568f50c48448079f7dfa0a147fada5-org6-peer6f0e.uk02.blockchain.ibm.com:31002 status: Error: 1 CANCELLED: Received http2 header with status: 503 {}$**

## Possible Fix
Check for the CANCELLED (1) code as well.
```
if (payload instanceof Error && payload.code && (payload.code === 14 || payload.code === 1)) {
    throw payload;
}

```
## Steps to Reproduce

Setup network in ibm cloud with multiple peers, stop first peer running try and ping network.

**Comment by crazyman1979** [comment]
@davidkel I hope you don't mind me tagging you, but i can see your PR here
https://github.com/hyperledger/composer/pull/3639/files/46d65f9000557fcd12a416406dc2784d6ac867a6

Is there any reason why retries only happen when code 14 (UNAVAILABLE) is seen
https://grpc.io/grpc/node/grpc.html

Would adding retries when other codes are seen make sense, for example DEADLINE_EXCEEDED.
Or maybe i'm missing something ...

**Comment by davidkel** [comment]
Thanks for the suggestion I will look to adding code=1 and also code 4 for deadline_exceeded to the 0.19.x code base. Unfortunately due to changes in the node sdk, these codes don't get returned anymore so cannot be applied to master and the 0.20 release.

**Comment by crazyman1979** [comment]
Thanks.
what happens now in 0.20 in relation to retries, i see this code hasn't changed.

**Comment by davidkel** [comment]
@crazyman1979 It probably won't work at the moment. The node-sdk changed how errors are reported and may not be possible to determine the difference between a network failure and a response from the peer. It needs more investigation.

**Comment by davidkel** [comment]
I have merged a fix for 0.19 and there is a separate issue for 0.20 under #4426
